server/src/main/java/org/apache/druid/segment/realtime/plumber/Plumbers.java

```diff
@@ -51,9 +53,9 @@ public static void addNextRow(
     try {
       inputRow = firehose.nextRow();
     }
-    catch (ParseException e) {
-      if (reportParseExceptions) {
-        throw e;
+    catch (ParseException | IOException e) {
```

Comment:
I think `IOException` is different in spirit from `ParseException` and should not be treated the same. Could you modify `Plumbers.addNextRow` to `throws IOException` instead of catching the `IOException` here?

Reply:
Ack, updated the PR. Much better indeed.